File: src/main/java/com/bvanseg/gigeresque/common/block/CustomStairsBlock.java
package com.bvanseg.gigeresque.common.block;

import net.minecraft.block.Block;
import net.minecraft.block.BlockState;
import net.minecraft.block.StairsBlock;

public class CustomStairsBlock extends StairsBlock {
	public CustomStairsBlock(BlockState baseBlockState, Settings settings) {
		super(baseBlockState, settings);
	}

	public CustomStairsBlock(Block block, Settings settings) {
		this(block.getDefaultState(), settings);
	}

	public CustomStairsBlock(Block block) {
		this(block, Settings.copy(block));
	}
}

File: src/main/kotlin/com/bvanseg/gigeresque/common/block/CustomStairsBlock.kt
package com.bvanseg.gigeresque.common.block

import net.minecraft.block.Block
import net.minecraft.block.BlockState
import net.minecraft.block.StairsBlock

/**
 * @author Boston Vanseghi
 */
class CustomStairsBlock(baseBlockState: BlockState, settings: Settings) : StairsBlock(baseBlockState, settings) {
    constructor(block: Block, settings: Settings) : this(block.defaultState, settings)
    constructor(block: Block) : this(block.defaultState, Settings.copy(block))
}